readers/mocks/messages.go

```diff
@@ -43,69 +43,95 @@ func (repo *messageRepositoryMock) ReadAll(chanID string, rpm readers.PageMetada
 	json.Unmarshal(meta, &query)
 
 	var msgs []readers.Message
+	filter := false
 	for _, m := range repo.messages[chanID] {
 		senml := m.(senml.Message)
+		filterOk := false
 
-		filter := false
+	forLoop:
```

Comment:
This looks unnecessarily complicated. Can you try to simplify with something like this:
```go

func (repo *messageRepositoryMock) ReadAll(chanID string, rpm readers.PageMetadata) (readers.MessagesPage, error) {
	repo.mutex.Lock()
	defer repo.mutex.Unlock()

	if rpm.Format != "" && rpm.Format != "messages" {
		return readers.MessagesPage{}, nil
	}

	var query map[string]interface{}
	meta, _ := json.Marshal(rpm)
	json.Unmarshal(meta, &query)

	var msgs []readers.Message
	for _, m := range repo.messages[chanID] {
		senml := m.(senml.Message)
		filter := true

		for name := range query {
			switch name {
			case "subtopic":
				if rpm.Subtopic != senml.Subtopic {
					filter = false
				}
			case "publisher":
				if rpm.Publisher != senml.Publisher {
					filter = false
				}
			case "name":
				if rpm.Name != senml.Name {
					filter = false
				}
			case "protocol":
				if rpm.Protocol != senml.Protocol {
					filter = false
				}
			case "v":
				if senml.Value != nil &&
					*senml.Value != rpm.Value {
					filter = false
				}
			case "vb":
				if senml.BoolValue != nil &&
					*senml.BoolValue != rpm.BoolValue {
					filter = false
				}
			case "vs":
				if senml.StringValue != nil &&
					*senml.StringValue != rpm.StringValue {
					filter = false
				}
			case "vd":
				if senml.DataValue != nil &&
					*senml.DataValue != rpm.DataValue {
					filter = false
				}
			case "from":
				if senml.Time < rpm.From {
					filter = false
				}
			case "to":
				if senml.Time > rpm.To {
					filter = false
				}
			}
			if !filter {
				break
			}
		}

		if filter {
			msgs = append(msgs, m)
		}
	}

	numOfMessages := uint64(len(msgs))

	if rpm.Offset >= numOfMessages {
		return readers.MessagesPage{}, nil
	}

	if rpm.Limit < 1 {
		return readers.MessagesPage{}, nil
	}

	end := rpm.Offset + rpm.Limit
	if end > numOfMessages {
		end = numOfMessages
	}

	return readers.MessagesPage{
		PageMetadata: rpm,
		Total:        uint64(len(msgs)),
		Messages:     msgs[rpm.Offset:end],
	}, nil
}

```

Reply:
I agree that it's complex , I'll try to simplify it but your example is worn because you only consider filtered messages.

Reply:
Sorry my bad, your example was wrong but because of `v`, `vb`, `vd` and `vs` checks (you can verify in my next commit) but not because of non filtered messages.